src/encoded/types/target.py

```diff
@@ -80,13 +86,19 @@ def __name__(self):
 
     def _name(self, properties):
         root = find_root(self)
-        organism = root.get_by_uuid(self.organism(properties=properties, return_uuid=True))
-        organism_props = organism.upgrade_properties()
-        return u'{}-{}'.format(properties['label'], organism_props['name'])
+        organism_uuid = self.organism(properties=properties, return_uuid=True)
+        if organism_uuid is None:
+            source = properties['investigated_as'][0].replace(' ',  '_')
```

Comment:
Do we always want first item, and is this deterministic ordering? I guess investigated_as is required, so this won't be empty at least.

Reply:
The way "investigated_as" work is if it has "synthetic tag", it cannot have other investigated_as and it won't have target_organism or genes thus no organism calculated. In another word, if a target doesn't have organism calculated, it should be `properties['investigated_as'] == ['synthetic tag']`.